lib/babe/build.go

```diff
@@ -180,42 +177,38 @@ func (b *Service) buildBlockBabeHeader(slot Slot) (*types.BabeHeader, error) {
 // buildBlockExtrinsics applies extrinsics to the block. it returns an array of included extrinsics.
 // for each extrinsic in queue, add it to the block, until the slot ends or the block is full.
 // if any extrinsic fails, it returns an empty array and an error.
-func (b *Service) buildBlockExtrinsics(slot Slot) ([]*transaction.ValidTransaction, error) {
+func (b *Service) buildBlockExtrinsics(slot Slot) []*transaction.ValidTransaction {
 	next := b.nextReadyExtrinsic()
 	included := []*transaction.ValidTransaction{}
 
 	for !hasSlotEnded(slot) && next != nil {
 		b.logger.Trace("build block", "applying extrinsic", next)
 		ret, err := b.rt.ApplyExtrinsic(next)
+
+		t := b.transactionState.Pop()
 		if err != nil {
-			return nil, err
+			continue
 		}
 
 		// if ret == 0x0001, there is a dispatch error; if ret == 0x01, there is an apply error
 		if ret[0] == 1 || bytes.Equal(ret[:2], []byte{0, 1}) {
 			errTxt, err := determineError(ret)
-			if err != nil {
-				return nil, err
-			}
 			// remove invalid extrinsic from queue
-			b.transactionState.Pop()
-
-			// re-add previously popped extrinsics back to queue
-			b.addToQueue(included)
-
-			return nil, errors.New("error applying extrinsic: " + errTxt)
+			if err == nil {
+				b.addToQueue(included)
+			}
```

Comment:
please remove this, if the extrinsic is invalid we shouldn't re-add it back to the queue